AEPCore/Sources/eventhub/ExtensionRuntime.swift

```diff
@@ -66,4 +66,22 @@ public protocol ExtensionRuntime {
     ///   - barrier: If true, the `EventHub` will only return `.set` if `extensionName` has moved past `event`
     /// - Returns: A `SharedStateResult?` for the requested `extensionName` and `event`
     func getSharedState(extensionName: String, event: Event?, barrier: Bool) -> SharedStateResult?
+
+    // MARK: - XDM Shared State
+
+    /// Creates a new XDM `SharedState` for this extension
+    /// If `event` is nil, one of two behaviors will be observed:
+    /// 1. If this extension has not previously published a shared state, shared state will be versioned at 0
+    /// 2. If this extension has previously published a shared state, shared state will be versioned at the latest
+    /// - Parameters:
+    ///   - data: Data for the `SharedState`
+    ///   - event: `Event` for which the `SharedState` should be versioned
+    func createXDMSharedState(data: [String: Any], event: Event?)
```

Comment:
do we need `createPendingXDMSharedState` ?

Reply:
I don't believe we have a use-case for pending XDM shared state. The Edge extension will just grab whatever is available for a given event.

@emdobrin can you confirm

Reply:
I think all of the requirements for shared state exist for the xdm counterpart.  Edge Extension is just one potential consumer of XDM Shared states, eventually we'd expect all shared states to be XDM -- as a concrete schema for all data will be super useful with rules and other cross-extension workflows.

Reply:
If we have the same functionality as regular shared states where you can fetch latest when pending, I think we should add pending states for important data that always needs to be up-to-date. The extensions will decide if they need to use pending for sync use-cases.

Reply:
Ok, the original spec didn't specify a pending XDM shared state API, but I'll update this PR to add one.

Reply:
@shalehaha @emdobrin added pending APIs.